source/postprocess/boundary_velocity_residual_statistics.cc

```diff
@@ -92,6 +116,26 @@ namespace aspect
 
 
 
+    template <int dim>
+    std::vector<Vector<double> >
+    BoundaryVelocityResidualStatistics<dim>::get_data_surface_strain_rate (const Point<dim> &p) const
+    {
+      Vector <double> temp_vector (dim-1);
+      std::vector<Vector<double> > data_surface_strain_rate (dim-1, temp_vector);
+      Point<dim> position = p;
+
+      if (this->get_geometry_model().natural_coordinate_system() == Utilities::Coordinates::spherical)
+        position = get_spherical_position(p);
+
+      for (unsigned int d=0; d<dim-1; ++d)
+        for (unsigned int e=0; e<dim-1; ++e)
+          data_surface_strain_rate[d][e] = strain_rate_data_lookup->get_data(position, d+e);
```

Comment:
I think this should be:

```suggestion
          data_surface_strain_rate[d][e] = strain_rate_data_lookup->get_data(position, d*(dim-1)+e);
```

Otherwise d=1,e=0 and d=0,e=1 would access the same data column (which in theory would be fine if the tensor is symmetric and the data file only stores 3 columns, but that does not seem to be the case).